chain/types/ethtypes/eth_transactions.go

```diff
@@ -89,12 +91,26 @@ func NewEthTxArgsFromMessage(msg *types.Message) (EthTxArgs, error) {
 		}
 		to = &addr
 
-		if len(msg.Params) > 0 {
-			params, err := cbg.ReadByteArray(paramsReader, uint64(len(msg.Params)))
+		if len(msg.Params) == 0 {
+			if msg.Method != builtintypes.MethodSend {
+				return EthTxArgs{}, xerrors.Errorf("cannot invoke method %d on non-EAM actor without params", msg.Method)
+			}
+		} else {
+			if msg.Method != builtintypes.MethodsEVM.InvokeContract {
+				return EthTxArgs{},
+					xerrors.Errorf("invalid methodnum %d: only allowed non-send method is InvokeContract(%d)",
+						msg.Method,
+						builtintypes.MethodsEVM.InvokeContract)
+			}
+
+			params, err = cbg.ReadByteArray(paramsReader, uint64(len(msg.Params)))
 			if err != nil {
-				return EthTxArgs{}, err
+				return EthTxArgs{}, xerrors.Errorf("failed to read params byte array: %w", err)
+			}
+
```

Comment:
We need to assert that `params` is non-empty here. Otherwise, this _should_ have been a method send and someone pulled a fast one.